app/models/reporting/temporal_report.rb

```diff
@@ -0,0 +1,99 @@
+module Reporting
+  class TemporalReport
+    include ActiveModel::Model
+
+    attr_reader :date, :report_type
+
+    def initialize(date:, report_type:)
+      @date = date
+      @report_type = Types::TemporalReportType[report_type]
+    end
+
+    def to_param
```

Comment:
Might be worth a comment here to explain that PARAM_FORMAT, and constants like these are defined in subclasses.

Reply:
Perhaps you could have:

```ruby
def to_param
  date.strftime(param_format)
end

def param_format
  raise 'Implement in subclass.'
end
```

And then in subclasses:
def param_format
  PARAM_FORMAT
end
```

That way the base has to know less about the concrete implementation in the subclass